### ISSUE TITLE
Logging groups via extensions

### ISSUE THREAD

**Issue by Xe** [shockkolate/arata#27]
As mentioned in #26, Commands are going to be getting Extensions for help or the like. I think a good thing to add to these would be logging group support for logging actions to a file or channel.

Here's a table of a few logging groups I've come up with:

| Name | Description |
| :-- | :-- |
| `request` | Requests of some kind |
| `permchange` | Permissions change of some kind |
| `auth` | Authentication of some kind |
| `assign` | An assignment action of some kind |
| `fail` | A failure due to lack of permissions |
| `other` | A generic change not otherwise specified by the logs |

Logging groups should also be able to be composed like the permissions system outlined in #23. This would let you have Vhost Requests get logged to `#vhost` with a group def like `HostServ:request`. Another idea would be to allow `auth` to be composable with `fail` to allow something on the order of defining `#services` as having `*:* *:auth:fail`

**Comment by ori-sky** [comment]
I agree that a syntax similar to that of the permissions system would be useful for categorizing log entries. However, composition like `*:* *:auth:fail` seems overly complex as a feature of the syntax when the same can be accomplished with proper usage of the syntax outlined for the permissions system.

I might be misunderstanding the proposed logging groups thing though; but I think it's too early at this stage of development--when only a handful of NickServ commands have been determined as of yet--to come up with categories for logging.

Also, you might be misunderstanding what exactly command extensions are. An extension is just an additional chunk of data associated with a `Command` so logging based on that data doesn't make sense as it would log **any** usage of the command, regardless of what actually happens.

Also, logging would be done by the Arata core as there is currently no mechanism for any sort of event firing and handling in plugin modules, so it wouldn't make sense to put that logging information into some extension when it probably does belong in `Command` itself.
